### PR TITLE
support https client-authentication

### PR DESCRIPTION
**Pull request by Blaimi**
add parameters --https-{cert,key}-file for https client-authentication.

### CODE REVIEW

File: apt_offline_core/AptOfflineCoreLib.py [hunk 1]
```diff
@@ -920,6 +923,14 @@ def fetcher( args ):
                         urllib.request.install_opener(opener)
                         log.verbose("Proxy successfully set up with Host %s and default port\n" % (Str_ProxyHost) )
         
+        if Str_HttpsCertFile and Str_HttpsKeyFile:
```
Comment (rickysarraf) [inline apt_offline_core/AptOfflineCoreLib.py]: Can we do a check for the file's path here ?

Comment (Blaimi) [inline apt_offline_core/AptOfflineCoreLib.py]: sounds good

**Comment by Blaimi** [comment]
I tested this against our own repository which identifies customers with client-certificates configured in Acquire::https::Ssl{Cert,Key}.

But I don't know, if this conflicts with the proxy-settings. I'm new to python and especially to urllib. We have a infrastructure to test all scenarios.

- https-authentication
  - trusted server-certificate
    - [ ] with proxy-server
    - [ ] without proxy-server
  - unstrusted server-certificate
    - [ ] with proxy-server
    - [ ] without proxy-server
- no https-authentication
  - trusted server-certificate
    - [ ] with proxy-server
    - [ ] without proxy-server
  - unstrusted server-certificate
    - [ ] with proxy-server
    - [ ] without proxy-server

As soon as our new Server goes to production, I can give you some Client-Certificates an signature-files, so you can test the implementation.

I don't get what you mean with username/password. The private key must not be encrypted by a password, that's true. we could add another parameter to get a password or document this as a requirement.

**Comment by rickysarraf** [comment]
Thanks. Once you give me some access, to run some tests, we are good to merge it in.

By username/password, I meant [simple Basic access authentication](https://en.wikipedia.org/wiki/Basic_access_authentication)

**Comment by Blaimi** [comment]
I sent you an email with a certificate to our update-server.

**Comment by rickysarraf** [comment]
Sorry for the very big delay. A lot many good events going on in personal life. I'm merging this for now, without testing. Since you've tested, this should just work. And it wouldn't be breaking any regular apt-offline use case. If it does, we can fix it.

Thank you for your patch, and being patient with me.

**Pull request merged**